packages/integration_test/lib/integration_test.dart

```diff
@@ -70,6 +70,22 @@ class IntegrationTestWidgetsFlutterBinding extends LiveTestWidgetsFlutterBinding
     };
   }
 
+  @override
+  void handlePointerEvent(
+    PointerEvent pointerEvent, {
+    TestBindingEventSource source = TestBindingEventSource.device,
+  }) {
+    if (kIsWeb) {
+      // Web driver does not create real device events. Routing web
+      // integration tests' pointer events with source set to `device`,
+      // results in exceptions.
+      super.handlePointerEvent(pointerEvent,
+          source: TestBindingEventSource.test);
```

Comment:
nit: trailing comma.

Why aren't we doing this in other test bindings? What's special about this one?

Reply:
I also added the other tests binding and update the comments. Fixing the assert statements didn't help since `WidgetTester.dispatchEvent` ([code](https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/widget_tester.dart#L786)) is not acting on the gesture bindings, using `TestBindingEventSource.device` does not generate the expected results for a gesture.

 `TestBindingEventSource.test` uses `RenderBinding.dispatchEvent`: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/binding.dart#L271